src/app/features/toolbar.component.ts

```diff
@@ -1,44 +1,70 @@
 const tmplToolbar: HTMLTemplateElement = document.createElement('template');
 tmplToolbar.innerHTML = `
 <app-header></app-header>
-<section class="bg-dark p-3 d-flex align-items-center justify-content-between">
-		<div class="d-flex gap-3">
-				<div class="bg-body rounded-circle">
-						<app-icon data-size="5rem" data-name="Eye"></app-icon>
-				</div>
-				<div class="d-flex justify-content-center flex-column">
-						<span class="text-white" data-i18n="profile"></span>
-						<span class="fs-4 fw-bold text-primary">Vision +</span>
-				</div>
-		</div>
-		<div class="d-grid gap-3 d-md-block">
-				<button type="button" class="btn btn-icon btn-inverse btn-secondary" data-i18n-title="openSettingsMode">
-						<span class="visually-hidden" data-i18n="openSettingsMode"></span>
-						<app-icon data-name="Settings"></app-icon>
-				</button>
-				<button type="button" class="btn btn-icon btn-inverse btn-secondary" data-i18n-title="pause">
-						<span class="visually-hidden" data-i18n="pause"></span>
-						<app-icon data-name="Pause"></app-icon>
-				</button>
-		</div>
-</section>
-
-<section class="d-flex flex-column p-3 mb-2">
-	<app-text></app-text>
-	<app-layout></app-layout>
-	<app-picture-video></app-picture-video>
-	<app-sound></app-sound>
-	<app-pointer></app-pointer>
-</section>
+
+<app-home class="d-none"></app-home>
+<app-modes class="d-none"></app-modes>
+<app-settings class="d-none"></app-settings>
+<app-edit-setting class="d-none"></app-edit-setting>
 `;
 
 class ToolbarComponent extends HTMLElement {
+	header: HTMLElement | null = null;
+	routeService: any;
 
 	constructor() {
 		super();
 
+		this.routeService = new routeService;
+
 		this.appendChild(tmplToolbar.content.cloneNode(true));
 	}
+
+	connectedCallback(): void {
+		this.header = this.querySelector('app-header');
+		this.routeService.initPages(this);
+		this.routeService.emitChangeEvent = (value: string) => {
+			this.setHeaderDisplay(value);
+		}
+
+		template.addEventListener('changeModeEvent', (event) => {
+			this.routeService.navigate(this.routeService.PAGE_MODES);
+		});
+		template.addEventListener('selectModeEvent', (event) => {
+			this.routeService.navigate(this.routeService.PAGE_HOME);
+		});
+		template.addEventListener('settingsEvent', (event) => {
+			this.routeService.navigate(this.routeService.PAGE_SETTINGS);
+		});
+		template.addEventListener('prevEvent', (event) => {
+			this.routeService.previous();
+		});
```

Comment:
Ça serait un bon cas pour utiliser `handleEvent` mais pas urgent, ton code semble fonctionnel et juste un peu répétitif.

Une source récente : https://gomakethings.com/the-handleevent-method-is-the-absolute-best-way-to-handle-events-in-web-components/